R/assign.R

```diff
@@ -0,0 +1,243 @@
+#' Derive an SDTM variable
+#'
+#' @description
+#' [sdtm_assign()] is an internal function packing the same functionality as
+#' [assign_no_ct()] and [assign_ct()] together but aimed at developers only.
+#' As a user please use either [assign_no_ct()] or [assign_ct()].
+#'
+#' @param raw_dat The raw dataset (dataframe); must include the
+#'   variables passed in `id_vars` and `raw_var`.
+#' @param raw_var The raw variable: a single string indicating the name of the
+#'   raw variable in `raw_dat`.
+#' @param tgt_var The target SDTM variable: a single string indicating the name
+#'   of variable to be derived.
+#' @param ct Study controlled terminology specification: a dataframe with a
+#'   minimal set of columns, see [ct_vars()] for details. This parameter is
+#'   optional, if left as `NULL` no controlled terminology recoding is applied.
+#' @param cl A code-list code indicating which subset of the controlled
+#'   terminology to apply in the derivation. This parameter is optional, if left
+#'   as `NULL`, all possible recodings in `ct` are attempted.
+#' @param tgt_dat Target dataset: a data frame to be merged against `raw_dat` by
+#'   the variables indicated in `id_vars`. This parameter is optional, see
+#'   section Value for how the output changes depending on this argument value.
+#' @param id_vars Key variables to be used in the join between the raw dataset
+#'   (`raw_dat`) and the target data set (`raw_dat`).
+#'
+#' @returns The returned data set depends on the value of `tgt_dat`:
+#' - If no target dataset is supplied, meaning that `tgt_dat` defaults to
+#' `NULL`, then the returned data set is `raw_dat`, selected for the variables
+#' indicated in `id_vars`, and a new extra column: the derived variable, as
+#' indicated in `tgt_var`.
+#' - If the target dataset is provided, then it is merged with the raw data set
+#' `raw_dat` by the variables indicated in `id_vars`, with a new column: the
+#' derived variable, as indicated in `tgt_var`.
+#'
+#'
+#' @importFrom rlang :=
+#' @keywords internal
+sdtm_assign <- function(raw_dat,
+                        raw_var,
+                        tgt_var,
+                        ct = NULL,
+                        cl = NULL,
```

Comment:
Does this name sound ok?
cod_dat <- Study codelist specification dataset (ct)
cod_lst --> Code List (cl)

Reply:
How far do you want to go with the renaming? I mean there are now several places in the code where `ct` and `cl` are used.

Reply:
All the places will be good. We can keep the parameter names consistent.

Reply:
What do you think about this?

cl_spec - Study codelist specification dataset (ct)
cl_name - Codelist name, example - C12345
read_cl_spec_example()
read_cl_spec()
assert_cl_spec()
assert_cl_name()